be/src/util/time_guard.h

```diff
@@ -46,6 +51,47 @@ class TimeGuard {
     int64_t _begin_time = 0;
 };
 
+// export from stack_util.h
+std::string get_stack_trace_for_thread(int tid, int timeout_ms);
+
+// SignalTimerGuard class manages a timer to capture and log thread stack traces after a specified timeout.
+// Note: If bthread yields. you may not get the expected stacktrace. But it's still safe.
+// usage:
+// {
+//     SignalTimerGuard guard(100);
+//     monitor_function();
+// }
+class SignalTimerGuard {
+public:
+    using TaskId = bthread::TimerThread::TaskId;
+    SignalTimerGuard(int64_t timeout_ms) {
+        if (timeout_ms > 0) {
+            auto timer_thread = bthread::get_global_timer_thread();
+            timespec tm = butil::milliseconds_from_now(timeout_ms);
+            int lwp_id = syscall(SYS_gettid);
+            void* arg = nullptr;
+            unaligned_store<int>(&arg, lwp_id);
+            _tid = timer_thread->schedule(dump_trace_info, arg, tm);
```

Comment:
will this cause invalid memory access because of the `arg` is allocated in stack and will be invalid after the constructor is done.

Reply:
This operation is passed the int parameter via void* and does not involve pointer manipulation.